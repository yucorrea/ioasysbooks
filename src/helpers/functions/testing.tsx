import React from 'react';
import renderer from 'react-test-renderer';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import theme from '../../config/theme';

export const matchSnapshotWithTheme = (component :  React.ReactElement) => {
  const tree = renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
};

export const renderWithTheme = (component : React.ReactElement, options?: RenderOptions) => {
  return render(<ThemeProvider theme={theme}> {component} </ThemeProvider>, options);
};
