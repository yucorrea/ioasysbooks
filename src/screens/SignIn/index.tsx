import React, { useState, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components/native';

import background from '../../assets/background.png';
import logo from './../../assets/Logo.png';

import { Input } from '../../components/Input';
import { LOGIN } from '../../store/slices/userSlice';

export function SignIn() {
  const dispatch = useDispatch();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = useCallback(() => {
    dispatch(LOGIN({email, password}));
  }, [email, password, dispatch]);

  return (
    <StyledContainer>
      <StyledForm>
        <StyledWrapper>
          <StyledLogo source={logo} />
          <StyledTitle>Books</StyledTitle>
        </StyledWrapper>

        <Input
          label="Email"
          placeholder="Seu e-mail"
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          value={email}
          onChangeText={e => setEmail(e)}
        />
        <Input
          label="Senha"
          secureTextEntry
          placeholder="********"
          enableButton
          value={password}
          onChangeText={e => setPassword(e)}
          onPress={handleLogin}
        />
      </StyledForm>
    </StyledContainer>
  );
}

const StyledContainer = styled.ImageBackground.attrs({
  source: background,
})`
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 16px;
`;

const StyledWrapper = styled.View`
  flex-direction: row;
  align-items: center;
  margin-bottom: 48px;
`;

const StyledLogo = styled.Image``;

const StyledTitle = styled.Text`
  margin-left: 16px;
  font-size: 28px;
  font-family: ${({ theme }) => theme.fonts.light};
  color: ${({ theme }) => theme.colors.background};
`;

const StyledForm = styled.View`
  height: 224px;
`;
