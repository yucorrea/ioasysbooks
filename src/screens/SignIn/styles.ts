import styled from 'styled-components/native';

import background from '../../assets/background.png';

export const Container = styled.ImageBackground.attrs({
  source: background,
})`
  width: 100%;
  height: 100%;

  justify-content: center;
  padding: 16px;
`;

export const Wrapper = styled.View`
  flex-direction: row;
  align-items: center;
  margin-bottom: 48px;
`;

export const Logo = styled.Image``;

export const Title = styled.Text`
  font-size: 28px;
  margin-left: 16px;
  font-family: ${({ theme }) => theme.fonts.light};
  color: ${({ theme }) => theme.colors.background};
`;

export const Form = styled.View`
  height: 224px;
`;
