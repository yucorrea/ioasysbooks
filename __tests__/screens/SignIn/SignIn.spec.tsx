import React from 'react';
import { matchSnapshotWithTheme, renderWithTheme } from '../../../src/helpers/functions/testing';

import SignIn from '../../../src/screens/SignIn';

describe('SignIn Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<SignIn />);
  });

  test('renders correctly', () => {
    renderWithTheme(<SignIn />);
  });
});
