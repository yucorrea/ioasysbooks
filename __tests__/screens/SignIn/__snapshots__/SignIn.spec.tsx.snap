// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Screen matches snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "height": 224,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 48,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/Logo.png",
          }
        }
        style={
          Array [
            Object {},
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Heebo-Light",
              "fontSize": 28,
              "marginLeft": 16,
            },
          ]
        }
      >
        Books
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.32)",
            "borderRadius": 4,
            "flexDirection": "row",
            "marginBottom": 16,
            "paddingBottom": 8,
            "paddingLeft": 13,
            "paddingRight": 16,
            "paddingTop": 8,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.5)",
                "fontFamily": "Heebo-Regular",
                "fontSize": 12,
                "width": "100%",
              },
            ]
          }
        >
          Email
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Seu e-mail"
          placeholderTextColor="rgba(255,255,255,0.5)"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Heebo-Regular",
                "fontSize": 16,
                "marginTop": 4,
                "width": "100%",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.32)",
            "borderRadius": 4,
            "flexDirection": "row",
            "marginBottom": 16,
            "paddingBottom": 8,
            "paddingLeft": 13,
            "paddingRight": 16,
            "paddingTop": 8,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.5)",
                "fontFamily": "Heebo-Regular",
                "fontSize": 12,
                "width": "100%",
              },
            ]
          }
        >
          Senha
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="********"
          placeholderTextColor="rgba(255,255,255,0.5)"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Heebo-Regular",
                "fontSize": 16,
                "marginTop": 4,
                "width": "100%",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        animatedStyle={
          Object {
            "value": Object {
              "height": 36,
              "width": 86,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "height": 36,
            "width": 86,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 44,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            animatedStyle={
              Object {
                "value": Object {
                  "opacity": 1,
                },
              }
            }
            collapsable={false}
            style={
              Object {
                "color": "#B22E6F",
                "fontFamily": "Heebo-Medium",
                "fontSize": 16,
                "opacity": 1,
              }
            }
          >
            Entrar
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
